example_test.go

```diff
@@ -0,0 +1,238 @@
+package schnorrkel_test
+
+import (
+	"fmt"
+
+	"github.com/ChainSafe/go-schnorrkel"
+	"github.com/gtank/merlin"
+)
+
+func ExampleMiniSecretKey() {
+	// To create a private-public keypair from a subkey keypair, use `NewMiniSecretKeyFromRaw`
+	// This example uses the substrate built-in key Alice:
+	// $ subkey inspect //Alice
+	priv, err := schnorrkel.NewMiniSecretKeyFromHex("0xe5be9a5092b81bca64be81d212e7f2f9eba183bb7a90954f7b76361f6edb5c0a")
+	if err != nil {
+		panic(err)
+	}
+
+	pub := priv.Public()
+	fmt.Printf("0x%x", pub.Encode())
+	// Output: 0xd43593c715fdd31c61141abd04a99fd6822c8558854ccde39a5684e7a56da27d
+}
+
+func ExampleMiniSecretKey_ExpandEd25519() {
+	msg := []byte("hello")
+	signingCtx := []byte("example")
+
+	signingTranscript := schnorrkel.NewSigningContext(signingCtx, msg)
+
+	msk, err := schnorrkel.NewMiniSecretKeyFromHex("0xe5be9a5092b81bca64be81d212e7f2f9eba183bb7a90954f7b76361f6edb5c0a")
+	if err != nil {
+		panic(err)
+	}
+
+	sk := msk.ExpandEd25519()
+
+	_, err = sk.Sign(signingTranscript)
+	if err != nil {
+		panic(err)
+	}
+
+	fmt.Println("expanded private key")
+	// Output: expanded private key
```

Comment:
```suggestion
```

Reply:
TIL about example tests, thanks! :) I thought it was used somewhere in CI but couldn't find anything. Turns out to be a language feature.